CMake/selxModules.cmake

```diff
@@ -155,34 +165,25 @@ macro( _selxmodule_target_file TARGETS )
   endforeach()
 endmacro()
 
-macro( _selxmodule_disable MODULE )
-  set( USE_${MODULE} FALSE )
-  list( FILTER SUPERELASTIX_INCLUDE_DIRS MATCHES EXCLUDE REGEX "(.*)${MODULE}(.*)" )
-  list( FILTER SUPERELASTIX_LIBRARY_DIRS MATCHES EXCLUDE REGEX "(.*)${MODULE}(.*)" )
-  list( FILTER SUPERELASTIX_LIBRARIES MATCHES EXCLUDE REGEX "(.*)${MODULE}(.*)" )
-  list( FILTER SUPERELASTIX_TEST_SOURCE_FILES MATCHES EXCLUDE REGEX "(.*)${MODULE}(.*)" )
-endmacro()
-
 # ---------------------------------------------------------------------
 # Public macros
 
 macro( enable_module MODULE )
   _selxmodule_enable( ${MODULE} "user" )
 endmacro()
 
-macro( enable_modules MODULES )
-  foreach( MODULE ${${MODULES}} )
-    enable_module( ${MODULE} )
+# Enable user-selected modules
+macro( enable_modules )
+  foreach( MODULE ${SUPERELASTIX_MODULES} )
+    if( USE_${MODULE} )
+      enable_module( ${MODULE} )
+    endif()
   endforeach()
+  foreach( MODULE ${SUPERELASTIX_MODULES} )
+    if( ${${MODULE}_IS_ENABLED})
```

Comment:
This if-statement could be part of the above foreach() and avoid redundant loop statement

Reply:
I think it cannot. Wait for my comments I just wrote in the commit I'm about to do :)